src/adrules/linsolve.jl

```diff
@@ -58,26 +58,28 @@ function ChainRulesCore.rrule(config::RuleConfig{>:HasReverseMode},
 
     # f defines a linear map => pullback defines action of the adjoint
     (y, f_pullback) = rrule_via_ad(config, f, x)
-    fᴴ(xᴴ) = f_pullback(xᴴ)[2]
+    fᴴ(xᴴ) = add(zerovector(x), f_pullback(xᴴ)[2])
```

Comment:
I'm still not sure about this way of implementing it. This is the equivalent of trying to reconstruct the vector type, but I feel like this should really be the responsability of the user to make sure that the rrule of the function outputs something sensible. Alternatively, it would be nice to have the option of specifying this manually

Reply:
I didn't realise that this was already changed. I think it's better to keep the old behaviour for now. The reason for the new implementation was that the tangent types of ChainRules are not necessarily compatible with VectorInterface, and this bypasses that by forcing the tangent type to be the same as the input vector type. However, it seems needlessly lossy to do the addition every time, allocating 2 extra vectors, so for now let's just assume that the user writes their code such that the AD rule of applying the operator yields a compatible vector type.